gateleen-cache/src/main/java/org/swisspush/gateleen/cache/storage/RedisCacheStorage.java

```diff
@@ -77,87 +77,93 @@ private long lockExpiry(long taskInterval) {
                 String message = "Failed to get cached request '" + cacheIdentifier + "'. Cause: " + logCause(event);
                 log.error(message);
-                future.fail(message);
+                promise.fail(message);
             } else {
                 if (event.result() != null) {
-                    future.complete(Optional.of(Buffer.buffer(event.result())));
+                    promise.complete(Optional.of(Buffer.buffer(event.result().toBytes())));
                 } else {
-                    future.complete(Optional.empty());
+                    promise.complete(Optional.empty());
                 }
             }
         });
-        return future;
+        return promise.future();
     }
 
     @Override
     public Future<Long> clearCache() {
-        Future<Long> future = Future.future();
+        Promise<Long> promise = Promise.promise();
         List<String> keys = Collections.singletonList(CACHED_REQUESTS);
         List<String> arguments = List.of(CACHE_PREFIX, "true");
-        ClearCacheRedisCommand cmd = new ClearCacheRedisCommand(clearCacheLuaScriptState, keys, arguments, redisClient, log, future);
+        ClearCacheRedisCommand cmd = new ClearCacheRedisCommand(clearCacheLuaScriptState, keys, arguments, redisAPI, log, promise);
         cmd.exec(0);
-        return future;
+        return promise.future();
     }
 
     @Override
     public Future<Long> cacheEntriesCount() {
-        Future<Long> future = Future.future();
-        redisClient.scard(CACHED_REQUESTS, reply -> {
+        Promise<Long> promise = Promise.promise();
+        redisAPI.scard(CACHED_REQUESTS, reply -> {
             if (reply.failed()) {
                 String message = "Failed to get count of cached requests. Cause: " + logCause(reply);
                 log.error(message);
-                future.fail(message);
+                promise.fail(message);
             } else {
-                future.complete(reply.result());
+                promise.complete(reply.result().toLong());
             }
         });
 
-        return future;
+        return promise.future();
     }
 
     @Override
     public Future<Set<String>> cacheEntries() {
-        Future<Set<String>> future = Future.future();
-        redisClient.smembers(CACHED_REQUESTS, reply -> {
+        Promise<Set<String>> promise = Promise.promise();
+        redisAPI.smembers(CACHED_REQUESTS, reply -> {
             if (reply.failed()) {
                 String message = "Failed to get cached requests. Cause: " + logCause(reply);
                 log.error(message);
-                future.fail(message);
+                promise.fail(message);
             } else {
-                JsonArray array = reply.result();
+                JsonArray array = new JsonArray();
+                reply.result().stream().parallel()
+                        .forEach(e -> {
+                            synchronized (array) {
+                                array.add(e);
```

Comment:
**WARN**: WTH?!? - Using a parallel stream and then using a synchronized block inside? -> That looks pretty much like a lot of nonsense.

Reply:
Yep, weird, will have a look